src/internal/_curry1.js

```diff
@@ -10,7 +10,7 @@ module.exports = function _curry1(fn) {
   return function f1(a) {
     if (arguments.length === 0) {
       return f1;
-    } else if (a != null && a['@@functional/placeholder'] === true) {
+    } else if (a instanceof Object && a['@@functional/placeholder'] === true) {
```

Comment:
I don't mind changing it, though I'm curious to know what the particular concerns of using `instanceof` here are.